python/ml4ir/applications/ranking/tests/data/csv/synthetic/feature_config.yaml

```diff
@@ -0,0 +1,114 @@
+query_key:
+  name: query_str
+  node_name: query_str
+  trainable: false
+  dtype: int64
+  log_at_inference: true
+  feature_layer_info:
+    type: numeric
+    shape: null
+  serving_info:
+    name: query_key
+    required: false
+    default_value: 0
+  tfrecord_type: context
+rank:
+  name: pos
+  node_name: pos
+  trainable: false
+  dtype: int64
+  log_at_inference: true
+  feature_layer_info:
+    type: numeric
+    shape: null
+  serving_info:
+    name: pos
+    required: true
+    default_value: 0
+  tfrecord_type: sequence
+label:
+  name: label
+  node_name: label
+  trainable: false
+  dtype: int64
+  log_at_inference: true
+  feature_layer_info:
+    type: numeric
+    shape: null
+  serving_info:
+    name: label
+    required: false
+    default_value: 0
+  tfrecord_type: sequence
+features:
+  - name: feat_0
+    node_name: feat_0
+    trainable: true
+    dtype: float
+    log_at_inference: false
+    feature_layer_info:
+      type: numeric
+      shape: null
+    serving_info:
+      name: feat_0
+      required: true
+      default_value: 0.0
+    tfrecord_type: sequence
+  - name: feat_1
+    node_name: feat_1
+    trainable: true
+    dtype: float
+    log_at_inference: false
+    feature_layer_info:
+      type: numeric
+      shape: null
+    serving_info:
+      name: feat_1
+      required: true
+      default_value: 0.0
+    tfrecord_type: sequence
+  - name: feat_2
+    node_name: feat_2
+    trainable: true
+    dtype: float
+    log_at_inference: false
+    feature_layer_info:
+      type: numeric
+      shape: null
+    serving_info:
+      name: feat_2
+      required: true
+      default_value: 0.0
+    tfrecord_type: sequence
+  - name: query_str
```

Comment:
@calliebradley [Minor] I would just remove `query_str` from the features section of the YAML just so we are compatible with ml4ir. `query_key` is automatically added to `all_features` and other dictionaries in the feature_config.

Reply:
Removed second `query_str` feature but also happy to change this to `query_key` if we make a multi-digit `query_key` so that it can test the resampling :)

Reply:
Oops meant to tag you @lastmansleeping

Reply:
No. This should be good for now. `query_str` should be unique as well, to an extent.